investment-app/TRINV.Api/TRINV.Infrastructure/Repositories/Repository.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace TRINV.Infrastructure.Repositories;
+
+using Application.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+public class Repository<T> : IRepository<T>
+    where T : class
+{
+    private readonly ApplicationDbContext _context;
+
+    public Repository(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<T>> GetAllAsync()
+    {
+        return await _context.Set<T>().ToArrayAsync();
```

Comment:
Better change it to ListAsync().
It easy to work with IList collections

Reply:
Done.